dev/publish_changelog.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Provides facilities for publishing changelogs to
+# our github.io site.
+
+import argparse
+import datetime
+import os
+import re
+import sys
+
+from spinnaker.run import check_run_quick
+
+# Path to the posts directory in the spinnaker.github.io site.
+POSTS_DIR = '_posts'
+
+class ChangelogPublisher():
+
+  def __init__(self, options):
+    self.__version = options.version
+    self.__githubio_repo_uri = options.githubio_repo_uri
+    self.__repo_name = ''
+    self.__changelog_gist_uri = options.changelog_gist_uri
+
+
+  def __checkout_githubio_repo(self):
```

Comment:
Good point. It feels like this function shouldn't really deal with the repo path at all.